glib/src/convert.rs

```diff
@@ -0,0 +1,331 @@
+        let result = unsafe {
+            ffi::g_convert_with_iconv(
+                str_.as_ptr(),
+                str_.len() as isize,
+                self.0,
+                &mut bytes_read,
+                &mut bytes_written,
+                &mut error,
+            )
+        };
+        if result.is_null() {
+            Err(CvtError::new(unsafe { from_glib_full(error) }, bytes_read))
+        } else {
+            let slice = unsafe { Slice::from_glib_full_num(result, bytes_written as usize) };
+            Ok((slice, bytes_read))
+        }
+    }
+    #[doc(alias = "g_iconv")]
+    pub fn iconv(
+        &self,
+        inbuf: Option<&[u8]>,
+        outbuf: Option<&mut [std::mem::MaybeUninit<u8>]>,
+    ) -> Result<(usize, usize, usize), IConvError> {
+        let input_len = inbuf.as_ref().map(|b| b.len()).unwrap_or_default();
+        let mut inbytes_left = input_len;
+        let mut outbytes_left = outbuf.as_ref().map(|b| b.len()).unwrap_or_default();
+        let mut inbuf = inbuf
+            .map(|b| mut_override(b.as_ptr()) as *mut c_char)
+            .unwrap_or_else(ptr::null_mut);
+        let mut outbuf = outbuf
+            .map(|b| b.as_mut_ptr() as *mut c_char)
+            .unwrap_or_else(ptr::null_mut);
+        let conversions = unsafe {
+            ffi::g_iconv(
+                self.0,
+                &mut inbuf,
+                &mut inbytes_left,
+                &mut outbuf,
+                &mut outbytes_left,
+            )
+        };
+        if conversions as isize == -1 {
+            let err = io::Error::last_os_error();
+            let code = err.raw_os_error().unwrap();
+            if code == libc::EILSEQ || code == libc::EINVAL {
+                Err(IConvError::WithOffset {
+                    source: err,
+                    offset: input_len - inbytes_left,
+                })
+            } else {
+                Err(err.into())
+            }
+        } else {
+            Ok((conversions, inbytes_left, outbytes_left))
+        }
+    }
+}
+
+impl Drop for IConv {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::g_iconv_close(self.0);
+        }
+    }
+}
+
+#[doc(alias = "g_locale_to_utf8")]
+pub fn locale_to_utf8(opsysstring: &[u8]) -> Result<(GString, usize), CvtError> {
+    assert!(opsysstring.len() <= isize::MAX as usize);
+    let mut bytes_read = 0;
+    let mut bytes_written = 0;
+    let mut error = ptr::null_mut();
+    let result = unsafe {
+        ffi::g_locale_to_utf8(
+            opsysstring.as_ptr(),
+            opsysstring.len() as isize,
+            &mut bytes_read,
+            &mut bytes_written,
+            &mut error,
+        )
+    };
+    if result.is_null() {
+        Err(CvtError::new(unsafe { from_glib_full(error) }, bytes_read))
+    } else {
+        Ok((unsafe { from_glib_full(result) }, bytes_read))
+    }
+}
+
+#[doc(alias = "g_filename_to_utf8")]
+pub fn filename_to_utf8(opsysstring: &[u8]) -> Result<(GString, usize), CvtError> {
```

Comment:
Not `&OsStr`?

Reply:
I suppose it should be `Path`, but also this is inconsistent with the rest of these functions unless this one is finished: https://github.com/gtk-rs/gtk-rs-core/pull/643

Reply:
That PR makes it look like this could be autogenerated though, so maybe just remove this one for now?

Reply:
I guess I should finish that PR first then :)

Reply:
Done

Reply:
Fixed but now it needs https://github.com/gtk-rs/gir/pull/1367

Reply:
Merged, can you update to that version?